torch/csrc/distributed/rpc/ProcessGroupAgent.cpp

```diff
@@ -53,17 +53,66 @@ Message deserialize(std::istream& is) {
   return Message(std::move(payload), std::move(tensors), type, id);
 }
 
+std::unordered_map<std::string, int> collectNames(
+    const std::string& workerName, std::shared_ptr<c10d::ProcessGroup>& pg) {
```

Comment:
Should we have some restrictions on the workerName? For ex: no non-ASCII chars, no special chars etc. A simple rule would be to just allow [A-Za-z0-9] and not allowing the first char to be a number.

Reply:
Sure, I will add restrictions on worker names.